synapse/rest/client/devices.py

```diff
@@ -91,7 +91,7 @@ def __init__(self, hs: "HomeServer"):
         self.auth_handler = hs.get_auth_handler()
 
     class PostBody(RequestBodyModel):
-        auth: Optional[AuthenticationData]
+        auth: Optional[AuthenticationData] = None
```

Comment:
This also looks like a behaviour change. IIRC In Pydantic 1.x,

- `foo: Optional[T]` means that `foo` **must** be explicitly provided as either a `T` or as `None`,
- `foo: Optional[T] = None` means that `foo` _may_ be provided (if so, it must be a `T` or None`); if not, it is treated as `None`.

Is this correct? Has this changed in Pydantic 2.x?

Reply:
The prose at https://docs.pydantic.dev/2.0/migration/#required-optional-and-nullable-fields suggests that how I thought it worked _is_ how it worked under V2. Will have to double check how it works in V1 here.

Reply:
Ah, very good point. I guess it depends on whether `auth: Optional[AuthenticationData]` should be `None` by default or not then?

Reply:
If I understand the Note in the link you posted correctly, then in v1 it defaulted to `None` and with v2 it no longer does, so it is correct to set it to `None` explicitly as default, as that is what is the default with v1:

```
A field annotated as typing.Optional[T] will be required, and will allow for a value of None. It does not mean that the field has a default value of None. (This is a breaking change from V1.)
```